src/core/qgsvector3d.h

```diff
@@ -135,12 +135,26 @@ class CORE_EXPORT QgsVector3D
                           v1.x() * v2.y() - v1.y() * v2.x() );
     }
 
-    //! Returns the length of the vector
+    /**
+     * Returns the length of the vector.
+     * \see lengthSquared()
+     */
     double length() const SIP_HOLDGIL
     {
       return sqrt( mX * mX + mY * mY + mZ * mZ );
```

Comment:
Just out of curiosity... wouldn't be useful to change this to `return sqrt( lengthSquared() );` to avoid code duplication, or such change could introduce performance issues?